### DISCUSSION TITLE
请问是否可以安装的时候，同时安装第三方包产出的 binary 等文件

### DISCUSSION THREAD

**Discussion by ivanallen** [xmake-io/xmake#4056] [General]
请问是否可以安装的时候，同时安装第三方包产出的 binary 等文件。

比如第三方包 spdk，它会生成一些工具，这些工具希望也可以被一起安装。

**Comment by waruqi** [comment]
xmake install ? 还是 包安装？如果是包安装，原本附带的工具就会编译安装，也可以自己 config 控制。。xmake install 目前只会导出target 依赖的动态库

**Comment by ivanallen** [comment]
xmake install
现在确实是只导出了动态库
但是也会有让 binary 导出的需求
或者有没有变通的方案？

**Reply by waruqi** [reply]
自己 after_install 导出

**Reply by ivanallen** [reply]
我也想这么干，但是遇到几个问题：

- after_install 是针对 target 的，感觉写在哪个 target 下都不合适。
- 我不知道第三方包的 bin path，没办法去 os.cp

比如

```lua
add_requires("spdk 1.0.0")

target("foo"):
   set_kind("static")
   add_files("*.cc")
   add_packages("spdk")

target("bar"):
   add_files("*.cc")
   add_deps("foo")
```

我知道 spdk 包在安装的时候，也自动安装了 binary:

![image](https://github.com/xmake-io/xmake/assets/12481610/ca97a644-d387-4a3e-9379-ea61d9f88a01)


after_install 加在哪里合适，怎么加，怎么找到这个 bin path 地址

**Comment by waruqi** [comment]
你也只能在 target 里面才能获取到 package bindir

```
local bindir = path.join(target:pkg("spdk"):installdir(), "bin")
```
